File: demo/02-grundlagen/note.py

input = input("Note = ")

# Validierung Eingabe
note = None
try:
    note = int(input)
except:
    print("Falsche Eingabe, bitte geben sie eine Zahl ein.")
    exit(-1)

# Validierung Zahlenbereich 1..6
if note < 1 or note > 6:
    print("Falsche Eingabe, bitte geben Sie einen Wert von 1..6 ein.")
    exit(-2)


# Ausgabe
print(note)

if note >=4:
    print("Test bestanden")
else:
    print("Test leider nicht bestanden")    

if note == 6:
    print("sehr gut")
elif note == 5:
    print("gut")
elif note == 4:
    print("genugend")
else:
    print("ungenügend")

File: src/chapter05/ctrl/Note.java
package chapter05.ctrl;

import util.Reader;

public class Note {

	public static void main(String[] args) {

		int note = Reader.readInt("Bitte geben sie eine Note zwischen 1-6 ein:");
		
		// Eingabe Validierung
		if (note <1 || note>6) {
			System.out.println("Keine gültiger Wert, bitte geben sie eine Zahl zwischen 1-6 ein!");
			System.exit(-1);
		}

		// Verarbeitung
		if (note >= 4) {
			System.out.println("Test bestanden");
		} else {
			System.out.println("Test leider nicht bestanden");			
		}
		
		switch (note) {
			case 6: 
				System.out.println("Note 6: sehr gut");
				break;
			case 5: 
				System.out.println("Note 5: gut");
				break;
			case 4: 
				System.out.println("Note 4: genügend");
				break;
		
			default:
				System.out.println("ungenügend");
		}
	}

}
